src/geometry/manifold/partition_of_unity.lean

```diff
@@ -4,69 +4,79 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Yury G. Kudryashov
 -/
 import geometry.manifold.bump_function
+import topology.partition_of_unity
 
 /-!
 # Smooth partition of unity
 
-In this file we define `smooth_bump_covering`, a structure that will be used to construct a smooth
-partition of unity. Namely, a `smooth_bump_covering` of a set `s : set M` is a collection of
-`smooth_bump_function`s such that their supports is a locally finite family of sets, and for each
-point `x ∈ s` there exists a bump function `f i` in the collection such that `f i =ᶠ[𝓝 x] 1`. This
-structure is the main building block in the construction of a smooth partition of unity (see TODO),
-and can be used instead of a partition of unity in some proofs.
+In this file we define two structures, `smooth_bump_covering` and `smooth_partition_of_unity`. Both
+structures describe coverings of a set by a locally finite family of supports of smooth functions
+with some additional properties. The former structure is mostly useful as an intermediate step in
+the construction of a smooth partition of unity but some proofs that traditionally deal with a
+partition of unity can use a `smooth_bump_covering` as well.
 
-We say that `f : smooth_bump_covering I s` is *subordinate* to a map `U : M → set M` if for each
+Given a real manifold `M` and its subset `s`, a `smooth_bump_covering I M s` is a collection of
+`smooth_bump_function`s `f i` indexed by `i : f.ι` such that
+
+* center of each `f i` belongs to `s`;
+* the family of sets `support (f i)` is locally finite;
+* for each `x ∈ s`, there exists `i : f.ι` such that `f i =ᶠ[𝓝 x] 1`.
+In the same settings, a `smooth_partition_of_unity ι I M s` is a collection of smooth nonnegative
+functions `f i : C^∞⟮I, M; 𝓘(ℝ), ℝ⟯`, `i : ι`, such that
+
+* the family of sets `support (f i)` is locally finite;
+* for each `x ∈ s`, the sum `∑ᶠ i, f i x` equals one;
+* for each `x`, the sum `∑ᶠ i, f i x` is less than or equal to one.
+
+We say that `f : smooth_bump_covering I M s` is *subordinate* to a map `U : M → set M` if for each
 index `i`, we have `closure (support (f i)) ⊆ U (f i).c`. This notion is a bit more general than
 being subordinate to an open covering of `M`, because we make no assumption about the way `U x`
 depends on `x`.
 
 We prove that on a smooth finitely dimensional real manifold with `σ`-compact Hausdorff topology,
-for any `U : M → set M` such that `∀ x ∈ s, U x ∈ 𝓝 x` there exists a `smooth_bump_covering I s`
-subordinate to `U`.
+for any `U : M → set M` such that `∀ x ∈ s, U x ∈ 𝓝 x` there exists a `smooth_bump_covering I M s`
+subordinate to `U`. Then we use this fact to prove a similar statement about smooth partitions of
+unity.
 
```

Comment:
Can you comment here on the slight API difference between the topological case (where the index set is given) and the smooth case (where the index set is part of the smooth bump covering)? In fact, the index set is part of the smooth bump covering, but it is given for a smooth partition of unity, so I wonder if one shouldn't make everything more uniform here (taking iota as a parameter in all cases).

Reply:
Unified the API, removed an `inhabited` instance. While it's possible to add this instance (with `ι = s`), the proof is nontrivial, and the added value is very small.